pkg/kubelet/kubelet_server_journal.go

```diff
@@ -0,0 +1,390 @@
+	if len(queries) > 0 {
+		for _, q := range queries {
+			// The presence of / or \ is a hint that the query is for a log file. If the query is for foo.log without a
+			// slash prefix, the heuristics will still return the file contents.
+			if strings.ContainsAny(q, `/\`) {
+				nlq.Files = append(nlq.Files, q)
+			} else if strings.TrimSpace(q) != "" { // Prevent queries with just spaces
+				nlq.Services = append(nlq.Services, q)
+			}
+		}
+	}
+
+	// Prevent specifying  an empty or blank space query.
+	// Example: kubectl get --raw /api/v1/nodes/$node/proxy/logs?query="   "
+	if ok && (len(nlq.Files) == 0 && len(nlq.Services) == 0) {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("query"), queries, "query cannot be empty"))
+	}
+
+	var sinceTime time.Time
+	sinceTimeValue := query.Get("sinceTime")
+	if len(sinceTimeValue) > 0 {
+		sinceTime, err = time.Parse(time.RFC3339, sinceTimeValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("sinceTime"), sinceTimeValue, "invalid time format"))
+		} else {
+			nlq.SinceTime = &sinceTime
+		}
+	}
+
+	var untilTime time.Time
+	untilTimeValue := query.Get("untilTime")
+	if len(untilTimeValue) > 0 {
+		untilTime, err = time.Parse(time.RFC3339, untilTimeValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("untilTime"), untilTimeValue, "invalid time format"))
+		} else {
+			nlq.UntilTime = &untilTime
+		}
+	}
+
+	var boot int
+	bootValue := query.Get("boot")
+	if len(bootValue) > 0 {
+		boot, err = strconv.Atoi(bootValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("boot"), bootValue, err.Error()))
+		} else {
+			nlq.Boot = &boot
+		}
+	}
+
+	var tailLines int
+	tailLinesValue := query.Get("tailLines")
+	if len(tailLinesValue) > 0 {
+		tailLines, err = strconv.Atoi(tailLinesValue)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("tailLines"), tailLinesValue, err.Error()))
+		} else {
+			nlq.TailLines = &tailLines
+		}
+	}
+
+	if len(allErrs) > 0 {
+		return nil, allErrs
+	}
+
+	return nlq, allErrs
+}
+
+func validateServices(services []string) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	var totalLength int
+	for _, s := range services {
+		totalLength += len(s)
+		if err := safeString(s); err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("query"), s, err.Error()))
+		}
+	}
+
+	maxTotalServiceNamesLength := 1000
+	if totalLength > maxTotalServiceNamesLength {
+		allErrs = append(allErrs, field.TooLongMaxLength(field.NewPath("query"), services, maxTotalServiceNamesLength))
+	}
+	return allErrs
+}
+
+func (n *nodeLogQuery) validate() field.ErrorList {
+	// This is to account for the case that query is empty and the caller is directly trying to query a file
+	// Example: kubectl get --raw /api/v1/nodes/$node/proxy/logs/foo.log
+	if len(n.Services) == 0 && len(n.Files) == 0 && n.options == (options{}) {
+		return field.ErrorList{}
+	}
```

Comment:
this shouldn't be a valid query-mode request... making the file-mode and query-mode paths independent means we can drop this